src/OpenTelemetry.AutoInstrumentation.NLogTarget/OpenTelemetryTarget.cs

```diff
@@ -0,0 +1,211 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using NLog;
+using NLog.Config;
+using NLog.Targets;
+using OpenTelemetry;
+using OpenTelemetry.AutoInstrumentation;
+using OpenTelemetry.AutoInstrumentation.Configurations;
+using OpenTelemetry.Logs;
+
+namespace OpenTelemetry.AutoInstrumentation.NLogTarget;
+
+[Target("OpenTelemetryTarget")]
+public sealed class OpenTelemetryTarget : TargetWithContext
+{
+    private static readonly ConcurrentDictionary<string, object> LoggerCache = new(StringComparer.Ordinal);
+    private static LoggerProvider? _loggerProvider;
+    private static Func<string?, object?>? _getLoggerFactory;
+
+    public OpenTelemetryTarget()
+    {
+        Layout = "${message}";
+    }
+
+    [RequiredParameter]
+    public string? Endpoint { get; set; }
+
+    public string? Headers { get; set; }
+
+    public bool UseHttp { get; set; } = true;
+
+    public string? ServiceName { get; set; }
+
+    [ArrayParameter(typeof(TargetPropertyWithContext), "attribute")]
+    public IList<TargetPropertyWithContext> Attributes { get; } = new List<TargetPropertyWithContext>();
+
+    [ArrayParameter(typeof(TargetPropertyWithContext), "resource")]
+    public IList<TargetPropertyWithContext> Resources { get; } = new List<TargetPropertyWithContext>();
+
+    public bool IncludeFormattedMessage { get; set; } = true;
+
+    public new bool IncludeEventProperties { get; set; } = true;
+
+    public new bool IncludeScopeProperties { get; set; } = true;
+
+    public bool IncludeEventParameters { get; set; } = true;
+
+    public int ScheduledDelayMilliseconds { get; set; } = 5000;
+
+    public int MaxQueueSize { get; set; } = 2048;
+
+    public int MaxExportBatchSize { get; set; } = 512;
+
+    protected override void InitializeTarget()
+    {
+        base.InitializeTarget();
+
+        if (_loggerProvider != null)
+        {
+            return;
+        }
+
+        var createLoggerProviderBuilderMethod = typeof(Sdk).GetMethod("CreateLoggerProviderBuilder", System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.NonPublic)!;
+        var loggerProviderBuilder = (LoggerProviderBuilder)createLoggerProviderBuilderMethod.Invoke(null, null)!;
+
+        loggerProviderBuilder = loggerProviderBuilder
+            .SetResourceBuilder(ResourceConfigurator.CreateResourceBuilder(Instrumentation.GeneralSettings.Value.EnabledResourceDetectors));
+
+        loggerProviderBuilder = loggerProviderBuilder.AddOtlpExporter(options =>
+        {
+            var endpoint = Endpoint;
```

Comment:
When using NLog Layout, then do like this:
```c#
var endpoint = RenderLogEvent(Endpoint, LogEventInfo.CreateNullEvent());
```

Reply:
resolved in [4401757](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/4371/commits/440175748c7fc3d7ee52f84af92a7d813686ed9d)